### PR TITLE
chore: add syntax for multi-way joins

### LINKED ISSUE #1612: Clearer error message for multi-way joins.
Currently, if a user tries to join multiple sources in a single statement, KSQL returns the error:

```
ServerError:io.confluent.ksql.parser.exception.ParseFailedException
Caused by: null
```

We should have a clearer error that explains that multi-way joins are not supported atm.

### PR DESCRIPTION
**Pull request by agavra**
fixes #1612

### Description 
Simply adds the syntax for multi-way joins, but still fails if users try to use it. The message, however, is now clearer:
```
Invalid join criteria specified; KSQL does not support multi-way joins.
```

### Testing done 

All existing tests pass, and I added a unit and QTT test to make sure that the new message is returned in the case of a multi-way join.

### Reviewer checklist
- [ ] Ensure docs are updated if necessary. (eg. if a user visible feature is being added or changed).
- [ ] Ensure relevant issues are linked (description should include text like "Fixes #<issue number>")

### CODE REVIEW

File: ksqldb-engine/src/main/java/io/confluent/ksql/planner/LogicalPlanner.java [hunk 1]
```diff
@@ -421,36 +421,37 @@ private PlanNode buildSourceNode() {
 
     final List<AliasedDataSource> sources = analysis.getFromDataSources();
 
-    final Optional<JoinInfo> joinInfo = analysis.getOriginal().getJoin();
-    if (!joinInfo.isPresent()) {
+    if (!analysis.isJoin()) {
```
[the remark this answers is not in the document]
Comment (agavra) [reply ksqldb-engine/src/main/java/io/confluent/ksql/planner/LogicalPlanner.java]: just thought it was easier to read, we already had an `isJoin` method (I just exposed it)

File: ksqldb-parser/src/main/java/io/confluent/ksql/parser/tree/JoinedSource.java [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+  public enum Type {
+    INNER("INNER"), LEFT("LEFT OUTER"), OUTER("FULL OUTER");
+
+    private final String formattedText;
+
+    Type(final String formattedText) {
+      this.formattedText = Objects.requireNonNull(formattedText, "formattedText");
+    }
+
+    public String getFormatted() {
```
[the remark this answers is not in the document]
Comment (agavra) [reply ksqldb-parser/src/main/java/io/confluent/ksql/parser/tree/JoinedSource.java]: uh no idea - i just copied this over from `Join` previously and I'd rather keep it like that for now to reduce jitter

File: ksqldb-engine/src/main/java/io/confluent/ksql/planner/LogicalPlanner.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (big-andy-coates) [reply ksqldb-engine/src/main/java/io/confluent/ksql/planner/LogicalPlanner.java]: I don't think it ends up with more readable code though...

```java
    if (!analysis.isJoin()) {
      return buildNonJoinNode(sources);
    }

    if (sources.size() == 1) {
      throw new IllegalStateException("Expected more than one source. Got " + sources.size());
    }

   // build join
```

The second `if` is superfluous if you know what `isJoin` is doing.  Can just be:

```java
    if ((sources.size() == 1) {
      return buildNonJoinNode(sources);
    }

   // build join
````

[the remark this answers is not in the document]
Comment (agavra) [reply ksqldb-engine/src/main/java/io/confluent/ksql/planner/LogicalPlanner.java]: 🤔 I'll see if I can make this peak readable in my next PR

**Pull request merged**